dangerzone/conversion/pixels_to_pdf.py

```diff
@@ -1,153 +0,0 @@
-                height = int(f.read().strip())
-            with open(rgb_filename, "rb") as rgb_f:
-                untrusted_rgb_data = rgb_f.read()
-            # The first few operations happen on a per-page basis.
-            page_size = len(untrusted_rgb_data)
-            total_size += page_size
-            with contextlib.redirect_stdout(io.StringIO()):
-                pixmap = fitz.Pixmap(
-                    fitz.Colorspace(fitz.CS_RGB),
-                    width,
-                    height,
-                    untrusted_rgb_data,
-                    False,
-                )
-            pixmap.set_dpi(DEFAULT_DPI, DEFAULT_DPI)
-            if ocr_lang:  # OCR the document
-                self.update_progress(
-                    f"Converting page {page_num}/{num_pages} from pixels to searchable PDF"
-                )
-                if int(fitz.version[2]) >= 20230621000001:
-                    page_pdf_bytes = pixmap.pdfocr_tobytes(
-                        compress=True,
-                        language=ocr_lang,
-                        tessdata=get_tessdata_dir(),
-                    )
-                else:
-                    # XXX: In PyMuPDF v1.22.5, the function signature of
-                    # `pdfocr_tobytes()` / `pdfocr_save()` was extended with an argument
-                    # to explicitly set the Tesseract data dir [1].
-                    #
-                    # In earlier versions, the PyMuPDF developers recommend setting this
-                    # path via the TESSDATA_PREFIX environment variable. In practice,
-                    # this environment variable is read at import time, so subsequent
-                    # changes to the environment variable are not tracked [2].
-                    #
-                    # To make things worse, any attempt to alter the internal attribute
-                    # (`fitz.TESSDATA_PREFIX`) makes no difference as well, when using
-                    # the OCR functions. That's due to the way imports work in `fitz`,
-                    # where somehow the internal `fitz.fitz` module is shadowed.
-                    #
-                    # A hacky solution is to grab the `fitz.fitz` module from
-                    # `sys.modules`, and set there the TESSDATA_PREFIX variable. We can
-                    # get away with this hack because we have a proper solution for
-                    # subsequent PyMuPDF versions, and we know that nothing will change
-                    # in older versions.
-                    #
-                    # TODO: Remove after oldest distro has PyMuPDF >= v1.22.5
-                    #
-                    # [1]: https://pymupdf.readthedocs.io/en/latest/pixmap.html#Pixmap.pdfocr_save
-                    # [2]: https://github.com/pymupdf/PyMuPDF/blob/0368e56cfa6afb55bcf6c726e7f51a2a16a5ccba/fitz/fitz.i#L308
-                    sys.modules["fitz.fitz"].TESSDATA_PREFIX = get_tessdata_dir()  # type: ignore [attr-defined]
```

Comment:
Done in fff0e88

Reply:
Right, thanks for catching that. Fixed in 529548e9bf6b88e1bf30a183cfe60ba52d9d5007, however it's not properly tested yet. I expect that we will catch similar issues to this one, once we test in all of our platforms.

Reply:
Yeap, let's keep it open until properly tested.

Reply:
> We need to keep this in the new version of the code especially now that we'll have older hosts.

We can resolve this thread now, since older hosts will actually use a vendored PyMuPDF library, which is new. Therefore, we can remove this fallback section.